File: android2_lesson01/src/java/java/com/example/ama/android2_lesson01/ui/base/Presenter.java
package com.example.ama.android2_lesson01.ui.base;

import com.example.ama.android2_lesson01.model.Note;

/**
 * Base interface for Presenter
 *
 * @param <T> view with which to work
 */

public interface Presenter<T extends NoteView> {

    void attachView(T view);

    void detachView();

    void deleteNote(Note note);


}

File: android2_lesson01/src/kotlin/java/com/example/ama/android2_lesson01/ui/base/Presenter.kt
package com.example.ama.android2_lesson01.ui.base

import com.example.ama.android2_lesson01.model.Note

/**
 * Base interface for Presenter
 *
 * @param <T> view with which to work
 */

interface Presenter<T : NoteView> {
    fun attachView(view: T)
    fun detachView()

    fun deleteNote(note: Note?)

}